### ISSUE TITLE
integrate litdata streaming dataloader into ultralytics

### ISSUE THREAD

**Issue by mohanreddypmr** [ultralytics/ultralytics#15982]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

need to integrate litdata streaming dataset and dataloader methods into ultralytics , by using this we can fetch aws s3 or azure blob training data easily in encrypted way.

### Use case

optimizing datasets for streaming directly from cloud storage. Work with remote data without local downloads with features like loading data subsets, accessing individual samples, and resumable streaming.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@mohanreddypmr thank you for your suggestion! Integrating a streaming dataloader for cloud storage is a great idea. We encourage you to submit a Pull Request (PR) with your proposed changes, and our team will review it.

**Comment by vsarkhel** [comment]
I guess this is different from using something like Torchdata and the Dataloader2 interface or Datapipes? The Pytorch lighting litdata library looks interesting but it abstracts away to much from the user.

**Comment by glenn-jocher** [comment]
@vsarkhel thank you for your input! The integration would indeed differ from using Torchdata or Datapipes, as it focuses on seamless cloud data streaming. If you have specific concerns or suggestions, feel free to share them.

**Comment by tchaton** [comment]
Hey everyone ! @mohanreddypmr @glenn-jocher  @vsarkhel @plashchynski 

We are very excited to share that we have added an integration between LitData + Ultralytics: https://github.com/Lightning-AI/litData/pull/651.

We are very keen to hear from you all ;) 

Best,
T.C

**Comment by glenn-jocher** [comment]
@tchaton that's fantastic news! Thank you and the Lightning AI team for developing this integration. We're excited to see how this enhances cloud data streaming capabilities for the Ultralytics community. @mohanreddypmr this directly addresses your feature request - please test it out and let us know how it works for your use case.

**Comment by Borda** [comment]
Yeah, would it also work with Lightning YOLO? 🐿️  https://github.com/Borda/lightning-YOLOs
